File: commands/dressSense.js
const { SlashCommandBuilder } = require('discord.js');
const https = require('https');
const { API_KEY } = require('../config.json');
const lat = 24.77;
const lon = 46.73;



module.exports = {
	data: new SlashCommandBuilder()
		.setName('wear')
		.setDescription('Replies with what you should wear based on current weather'),
	async execute(interaction) {
        https.get(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=lat=${lon}&appid=${API_KEY}&units=metric`);
        let data = '';
        let weather = '';
        https.get(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${API_KEY}&units=metric`, (res) => {
        
            res.on('data', (d) => {
                data += d;
              //process.stdout.write(d);
              //process.stdout.write(d);
              
            });

            res.on('end', async() => {
                const body = JSON.parse(data);
                let wear = "I recommend you wear:\n";

                console.log(body);
              
                
                if (body.main.temp >= 22) {
                  wear = wear + ("- a short sleeved shirt \n") + ("- shorts \n");

                    //Check if it is cloudy/sunny 
                    if (body.clouds.all<40) {
                      wear = wear + ('- a hat \n') + ('- sunglasses \n');
                    }
                }

                if (body.main.temp >= 15 & body.main.temp <= 21) {
                  wear = wear + ('- a T-shirt \n');
                    checkIfWindy();
                }

                if (body.main.temp >= 7 & body.main.temp <= 14) {
                  wear = wear + ('- a hoodie or a sweater \n')
                    checkIfWindy();
                }
                
                if (body.main.temp >= -13 & body.main.temp <= 6) {
                  wear = wear + ('- a light or medium jacket \n');
                
                  if (body.wind.speed >= 40) {
                    wear = wear + ('- a hat\n');
                  }
                  
                }
                
                if (body.main.temp <= -14) {
                  wear = wear + ('- a winter jacket \n' ) + ('- a scarf \n') + ('- gloves \n') + ('- a hat \n ');
                }
                

                //Checks if it is raining
                if (body.weather[0].main.includes("rain") || body.weather[0].main.includes("Rain")) {
                  wear = wear + ('- a raincoat \n') + ('- rain boots \n') + ('- an umbrella \n');
              
                }

                console.log('wear:');
                console.log(wear);
                console.log(body.clouds.all);
                
                //Reply
                await interaction.reply({content: wear, ephemeral: true});

              //Check if it is windy
              function checkIfWindy() {
                if (body.wind.speed >= 40) {
                  wear = wear + ('- a windbreaker \n') + ('- a hat\n');
                }
              }


            });
          }).on('error', (e) => {
            console.error(e);
          });
    
	},
};

File: web/scripts/dressSense.py
# importing requests and json
import requests, json
from scripts import rewordSentence

with open('../config.json', 'r') as configfile:
    config = json.load(configfile)
    
with open('./storage/data_location.json', 'r') as locationJson:
    location = json.load(locationJson)
    
url1 = "https://api.openweathermap.org/data/2.5/weather?lat="

#Default lat and lon is Kelowna
lat = "49.88"
lon = "-119.47"
API_KEY = config['API_KEY']

URL = url1 + lat + "&lon=" + lon + "&appid=" + API_KEY + "&units=metric"


response = requests.get(URL)

#Grab current users latitude and longitude

def getLocation():
    lat = location[0]['latitude']
    lon = location[0]['latitude']


def getDressSense():
    getLocation()
    
    if response.status_code == 200: 
        data = response.json()
        main = data['main']
   
        temperature = main['temp']
        clouds = data['clouds']['all']
        windSpeed = data['wind']['speed']
        report = data['weather']
        isRaining = "rain" in report[0]['description']
        
        
        
        w = "I recommend you wear: \n"
        try:
            wear = rewordSentence.getRewordSentence(w) + "\n"
        except:
            wear = "I recommend you wear:  \n"
            print('there was an error with running rewordSentence')
            
   
        def checkIfWindy():
            if (windSpeed):
                 wear = wear + (' - a windbreaker \n') + ('- a hat\n')
           
   
        if(isRaining):
            wear = wear + ('- a raincoat \n') + ('- rain boots \n') + ('- an umbrella \n')
   
        if(temperature < -13):
            wear = wear + ('- a winter jacket \n' ) + ('- a scarf \n') + ('- gloves \n') + ('- a hat \n ')
   
        if(temperature >= -13 and temperature < 7):
            wear = wear + ('- a light or medium jacket \n')
            if(windSpeed >= 40):
                wear = wear + ('- a hat\n')
   
        if(temperature >= 7 and temperature < 15):
            wear = wear + ('- a hoodie or a sweater \n')
            checkIfWindy()
   
        if(temperature >= 15 and temperature < 22):
            wear = wear + ('- a T-shirt \n')
            checkIfWindy()
       
   
        if(temperature >= 22):
            wear = wear + ("- a short sleeved shirt \n") + ("- shorts \n")
            if(clouds<40):
                 wear = wear + ('- a hat \n') + ('- sunglasses \n')
                 
        try:
            wear = wear + sentimentNLTK.isNegOrIsPos(wear)
        except:
            print('there was an error with running sentimentNLTK')
         
                 
        return wear
    else:
        return("Sorry there was an error")
    
# print(getDressSense())
 
